src/bvh/bvh.rs

```diff
@@ -113,22 +137,37 @@ impl BVHNode {
         let mut buckets = [Bucket::empty(); NUM_BUCKETS];
         let mut bucket_assignments: [Vec<usize>; NUM_BUCKETS] = Default::default();
 
-        // Assign each shape to a bucket
-        for idx in &indices {
-            let shape = &shapes[*idx];
-            let shape_aabb = shape.aabb();
-            let shape_center = shape_aabb.center();
-
-            // Get the relative position of the shape centroid [0.0..1.0]
-            let bucket_num_relative = (shape_center[split_axis] - centroid_bounds.min[split_axis]) /
-                                      split_axis_size;
+        if split_axis_size < EPSILON {
+            // Spread the remaining shapes evenly across buckets,
+            // instead of using a heuristic
+            let mut bucket_num = 0;
+            for idx in &indices {
```

Comment:
You can use enumerate to get the index of `idx` in `&indIces` and use it to determine the bucket number
instead of using the mutable `bucket_num` approach.